source/guidelines/sdk-development.rst

```diff
@@ -64,215 +68,142 @@ Reactive
 
-Particular decisions to consider:
+You will find in the implementations different invariants to ensure the object is well constructed and a nicer API is published.
+
+Particular decisions we considered:
 
 * UInt64 support: While `Java supports big numbers <https://docs.oracle.com/javase/7/docs/api/java/math/BigInteger.html>`_, for example, JavaScript doesn't. The JavaScript SDK has a custom class to handle the `uint64 types <https://github.com/nemtech/nem2-sdk-typescript-javascript/blob/master/src/model/UInt64.ts>`_. If your language supports ``uint64``, use that implementation instead.
-*  API conversions: Sometimes, the data returned by API is compressed. You might need to convert those types for the user.
+* API conversions: Sometimes, the data returned by API is compressed. You might need to convert those types for the user.
 * `Namespace <https://github.com/nemtech/nem2-sdk-typescript-javascript/blob/master/src/model/namespace/NamespaceId.ts>`_ id: At creation time you add the string name, but when you receive the Namespace from the network, it comes in formatted as ``uint64`` id. A specific endpoint returns the Namespace ``string`` name.
 
 Transaction Serialization
 =========================
 
 The `catbuffer library <https://github.com/nemtech/catbuffer>`_ defines the protocol to serialize and deserialize |codename| entities.
 
-In combination with the catbuffer-generators project, developers can generate builder classes for a given set of programming languages.
-For example, the |sdk| uses the generated code to operate with the entities in binary form before announcing them to the network.
+In combination with the `catbuffer-generators project <https://github.com/nemtech/catbuffer-generators/tree/master/generators>`_, developers can generate builder classes for a given set of programming languages.
+For example, the |sdk| uses the generated code to operate with the entities in binary form.
+
+.. note:: If there is no generator for the programming language selected, you will need to develop it first. You can base your work on the `generator <https://github.com/nemtech/catbuffer-generators/tree/master/generators/typescript>`_ for TypeScript.
+
+If there is a generator, follow the next steps to generate the builders for all the existent entities:
+
+1. Clone the catbuffer-generators repository.
+
+.. code-block:: bash
+
+   git clone https://github.com/nemtech/catbuffer-generators
+
+2. Install the package requirements.
+
+.. code-block:: bash
+
+   pip install -r requirements.txt
+
+3. Clone the catbuffer repository inside the ``catbuffer-generators`` folder.
+
+.. code-block:: bash
+
+   git clone https://github.com/nemtech/catbuffer
+
+4. Generate code for all the schemas running the following command under the ``catbuffer-generators`` directory, replacing ``cpp_builder`` for the targeted programming language.
+
+.. code-block:: bash
+
+   python ../scripts/generate_all.sh cpp_builder
```

Comment:
Current generate_all.sh doesn't generate all the transactions at once creating incomplete files like `EntityTypeDto`.

I would recommend following the specific language generate script like:

https://github.com/nemtech/catbuffer-generators/blob/master/scripts/generate_java.sh
https://github.com/nemtech/catbuffer-generators/blob/master/scripts/generate_typescript.sh

Where the script also builds, tests and deploys the artifact for the given language.

The script should be added to [travis](https://github.com/nemtech/catbuffer-generators/blob/master/.travis.yml)

Reply:
Ideally, ``generate_all.sh`` should generate all the tranasctions. This note will really helpful for the ``catbuffer-generator`` README.